internal/controller/kubeletconfig_controller.go

```diff
@@ -155,9 +159,22 @@ func (r *KubeletConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			return true
 		}
 	}
+	numaResourcesOperatorPredicate := predicate.Funcs{
+		UpdateFunc: func(e event.TypedUpdateEvent[client.Object]) bool {
+			nroOld := e.ObjectOld.(*nropv1.NUMAResourcesOperator)
+			nroNew := e.ObjectNew.(*nropv1.NUMAResourcesOperator)
+			oldNodeGroups := nroOld.Spec.NodeGroups
+			newNodeGroups := nroNew.Spec.NodeGroups
+			// if nodeGroups number has changed,
+			// it means that we should iterate and create/delete ConfigMap data for RTE pods
+			return len(oldNodeGroups) != len(newNodeGroups)
```

Comment:
what about the case where the NROP CR was updated with different nodegroup but maintained same amount of entries in spec.nodegroup? in such case the condition would be false but in reality the groups changed

Reply:
Right. let's do a simple compare instead